### PR TITLE
test(webkit): add new scrolling tests to ensure correct webkit behavior

### PR DESCRIPTION
**Pull request by JoelEinbinder**
scrollable2 is just there so I can navigate between two unique but same-origin pages that have scrollbars. Our server doesnt support query params.

### CODE REVIEW

File: test/browsercontext-device.spec.ts [hunk 1]
```diff
@@ -58,4 +59,76 @@ describe('device', (suite, { browserName }) => {
...
   });
+
+  it('should scroll twice when emulated',(test, {browserName}) => {
+    test.skip(browserName === 'webkit');
+  }, async ({server, contextFactory, playwright, contextOptions}) => {
+    const device = playwright.devices['iPhone 6'];
+    const context = await contextFactory({
+      ...contextOptions,
+      ...device,
+      viewport: {
```
Comment (dgozman) [inline test/browsercontext-device.spec.ts:70]: Why both device and viewport? Here and everywhere.

File: test/browsercontext-device.spec.ts [hunk 2]
```diff
@@ -58,4 +59,76 @@ describe('device', (suite, { browserName }) => {
...
+          <button>hi</button>
+  `);
+    await new Promise(x => setTimeout(x, 500));
```
Comment (dgozman) [inline test/browsercontext-device.spec.ts:81]: Why this one?

File: test/browsercontext-device.spec.ts [hunk 3]
```diff
@@ -58,4 +59,76 @@ describe('device', (suite, { browserName }) => {
...
+    const page = await context.newPage();
+    await page.goto(server.PREFIX + '/input/scrollable.html');
+    await page.evaluate(() => window.scroll(0, 100));
+
+    await page.goto(server.PREFIX + '/input/scrollable2.html');
+    expect(await page.evaluate(() => window.scrollY)).toBe(0);
+    await context.close();
+  });
+
+  it('should scroll to a precise position with mobile scale', (test, {browserName}) => {
```
Comment (dgozman) [inline test/browsercontext-device.spec.ts:114]: This test seems very similar to the previous one. Let's combine them?

**Review by dgozman — APPROVED** [review]

File: test/browsercontext-device.spec.ts (hunk 1 above)
Comment (JoelEinbinder) [reply test/browsercontext-device.spec.ts]: Video needed even viewport demensions for when I was debugging. Removed.

File: test/browsercontext-device.spec.ts (hunk 2 above)
Comment (JoelEinbinder) [reply test/browsercontext-device.spec.ts]: Forgot about it. removed.

File: test/browsercontext-device.spec.ts (hunk 3 above)
Comment (JoelEinbinder) [reply test/browsercontext-device.spec.ts:103]: They test distinct issues which are not going to be solved with the same patch, so I'd prefer they remain separate.

**Pull request merged**